### PR TITLE
[T2 Port] Add listener details

### PR DESCRIPTION
**Pull request by alrod**
### Issue describing the changes in this PR

Port from track2:
https://github.com/Azure/azure-sdk-for-net/pull/25566

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [] My changes **should not** be added to the release notes for the next release
  * [x] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs [hunk 1]
```diff
@@ -73,6 +74,26 @@ internal sealed class ServiceBusListener : IListener, IScaleMonitorProvider
...
+                else if (_receiver.IsValueCreated)
+                {
+                    endpoint = _receiver.Value.ServiceBusConnection.Endpoint.ToString();
+                } 
+                else if (_sessionClient.IsValueCreated)
+                {
+                    endpoint = _sessionClient.Value.ServiceBusConnection.Endpoint.ToString();
+                }
+
+                return $"namespace='{endpoint}', enityPath='{_entityPath}', singleDispatch='{_singleDispatch}', " +
```
Comment (hossam-nasr) [inline src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs:94]: Typo

```suggestion
                return $"namespace='{endpoint}', entityPath='{_entityPath}', singleDispatch='{_singleDispatch}', " +

```

File: src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs [hunk 2]
```diff
@@ -73,6 +74,26 @@ internal sealed class ServiceBusListener : IListener, IScaleMonitorProvider
...
+                    endpoint = _clientEntity.ServiceBusConnection.Endpoint.ToString();
+                } 
+                else if (_receiver.IsValueCreated)
+                {
+                    endpoint = _receiver.Value.ServiceBusConnection.Endpoint.ToString();
+                } 
+                else if (_sessionClient.IsValueCreated)
+                {
+                    endpoint = _sessionClient.Value.ServiceBusConnection.Endpoint.ToString();
+                }
```
Comment (hossam-nasr) [inline src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs:92]: Do you mind clarifying why this is necessary here but wasn't in track2?

Comment (alrod) [reply src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs:92]: For track1 it's little bit messy as we have 3 different clients for 3 different scenarios, for track2 it's only one client

Comment (hossam-nasr) [reply src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs:92]: Thanks :)

**Review by hossam-nasr — APPROVED** [review]

**Pull request merged**